### ISSUE TITLE
testTask doesn't work under namespace

### ISSUE THREAD

**Issue by akoltun** [jakejs/jake#349]
If the test task is declared inside the namespace 

```javascript
namespace('test', function() {
  testTask('test inside namespace', function() {
    this.testName = 'task'
    this.testFiles.include(<test files>);
  });
});
```

the error arises:

```
TypeError: Cannot read property 'on' of undefined
```

because method `task` in the `test_task.js` module addresses the test task using `self.testName`. 

```javascript
  task(this.testName, prereqs, {async: true}, function () {
    var t = jake.Task[self.testName + ':run'];
    t.on('complete', function () {
      complete();
    });
    // Pass args to the namespaced test
    t.invoke.apply(t, arguments);
  });
```

**Comment by akoltun** [comment]
Pull request #350 solves this issue

**Comment by mde** [comment]
Published fix to NPM, v8.0.18.
